lib/junk_drawer/notifier.rb

```diff
@@ -29,7 +29,13 @@ def call(*args)
   private
 
     def strategy
-      STRATEGIES.fetch(self.class.strategy)
+      chosen_strategy = self.class.strategy
+
+      if chosen_strategy.is_a?(Class) && chosen_strategy.respond_to?(:call)
+        return chosen_strategy
+      end
+
+      STRATEGIES.fetch(chosen_strategy)
```

Comment:
I'm wondering if we might want to move this to the setter instead:

```rb
class << self
  attr_reader :strategy

  def strategy=(strategy)
    @strategy = strategy.is_a?(Symbol) ? STRATEGIES.fetch(strategy) : strategy
  end
end
```

Just a thought. I don't have strong opinion one way or another.

Reply:
oh yeah, that does feel cleaner. It also takes away the extra bit of complexity/confusion added by the actual strategy getting chosen on `#call` rather than when it's set.